receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -245,7 +249,7 @@ func (jr *jReceiver) SubmitBatches(ctx thrift.Context, batches []*jaeger.Batch)
 			ok = true
 			jr.nextProcessor.ProcessTraceData(ctx, td)
 			// We MUST unconditionally record metrics from this reception.
-			spansMetricsFn(td.Node, td.Spans)
+			internal.RecordTraceReceiverMetrics(ctxWithReceiverName, len(batch.Spans), len(batch.Spans)-len(td.Spans))
```

Comment:
not from your change, so let's do later but "internal" is not a good name for a package, it is part of its location on the repo but then it should be observability or something else.

Reply:
also not from your change directly but my thougts is that we should record all spans received and separately the dropped ones. I mean if the translator report an error we don't record those spans as received, and we just take into account the ones that were translated successfully. In practice there should be very translation issues but the metric of received and dropped will be incorrect on that case

Reply:
Not sure I understand what you mean. Dropped is intended to measure only the spans dropped by this receiver and the only reason is via translation.

If for every step (block, processor, however we call it) we record the number of received and dropped we will have a very good image about what is happening.

Reply:
I mean that when SubmitBatches is called we already received the spans, they should count as received. Right now if `ThriftBatchToOCProto` returns error those spans are not going to be counted at all (nor as received or dropped).

Reply:
I think I got what you want, but this was an error before.

Reply:
Added TODOs in places where we should also record metrics, also I recorded metrics where data were available.